src/ClientData/CaptureData.cpp

```diff
@@ -29,7 +29,7 @@ CaptureData::CaptureData(const CaptureStarted& capture_started,
                          std::optional<std::filesystem::path> file_path,
                          absl::flat_hash_set<uint64_t> frame_track_function_ids,
                          DataSource data_source)
-    : memory_sampling_period_ns_(capture_started.capture_options().memory_sampling_period_ns()),
+    : capture_started_(capture_started),
```

Comment:
Nit: I would recommend to change the signature of the constructor and have the object been taken by value and then move it into place.